src/DataObject/DataObjectDocument.php

```diff
@@ -157,7 +135,7 @@ public function shouldIndex(): bool
         }
 
         // Dataobject is only in draft
-        if ($dataObject->hasExtension(Versioned::class) && !$dataObject->isLiveVersion()) {
+        if ($dataObject->hasExtension(Versioned::class) && !$dataObject->isPublished()) {
```

Comment:
I will have a retest of this one, but I recall that I was getting issues with Pages not submitting to the Index, even though they were published, and even though there was new (live) content to be updated.

EG: You could publish the page and then very quickly edit something for Draft only, and since this Job is delayed, `isLiveVersion()` could end up returning `false`.

I'll check out that `isOnDraftOnly()` method as well. I think, ultimately, I don't want this method trying to figure out if there are new changes (it's not sophisticated enough). Something/somewhere triggered the desire to index, so it should just respect that and only use this as a sanity check that the DO is available to live users.

Reply:
I've re-tested this one. It wasn't actually an issue with my (real) application, it was an issue in unit tests - but this was because you actually have to re-fetch your records after publishing them, otherwise the (already instantiated) `Version` on your `DataObject` won't match the (dynamically fetched) `_Live::Version`.

https://github.com/silverstripe/silverstripe-search-service/pull/76/commits/284c171cd0e724ae0e30e553e15652f111be65d8